lib/index.js

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (c) 2017 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+var _ = require('lodash');
+var async = require('async');
+var bedrock = require('bedrock');
+var crypto = require('crypto');
+var jwt = require('jsonwebtoken');
+var database = require('bedrock-mongodb');
+var BedrockError = bedrock.util.BedrockError;
+
+var _state = null;
+
+// load config defaults
+require('./config');
+
+var api = {};
+module.exports = api;
+
+var logger = bedrock.loggers.get('app');
+
+// TODO: make responding to this event optional
+bedrock.events.on(
+  'bedrock-authn-did-jwt.config.secretStore', (strategy, callback) => {
+    strategy.setStore(api, callback);
+  });
+
+bedrock.events.on('bedrock-mongodb.ready', callback => {
+  async.waterfall([
+    callback => {
+      database.openCollections(['jwtSecrets'], callback);
+    },
+    callback => {
+      database.createIndexes([{
+        collection: 'jwtSecrets',
+        fields: {id: 1},
+        options: {unique: true, background: false}
+      }], callback);
+    }], callback);
+});
+
+/**
+ * Initialize a namespaced keystore.
+ *
+ * @param options the options to use
+ *   algorithm the JWT signing algorithm to use
+ *   clockToleranceInSecs the clockTolerance to use in during JWT verification
+ *   namespace the namespace to use
+ *   provision true for provisioning operation
+ *   ttlInSecs the JWT ttl
+ * @param callback(err, state) called once the operation completes.
+ */
+api.provision = function(options, callback) {
+  options = _.assign({}, options, {provision: true});
+  _getState(options, err => callback(err));
+};
+
+/**
+ * Create a JWT
+ *
+ * @param options the options to use
+ *   namespace the namespace for the keystore
+ *   payload the payload to sign and include in the JWT
+ *
+ * @param callback(err, state) called once the operation completes.
+ */
+api.sign = function(options, callback) {
+  async.auto({
+    getState: function(callback) {
+      return _getState({namespace: options.namespace}, callback);
+    },
+    create: ['getState', function(callback, results) {
+      var state = results.getState;
+      var nowInSecs = Math.floor(Date.now() / 1000);
+      var notAfter = (nowInSecs + state.tokenTtlInSecs);
+      var payload =
+        _.assign({}, options.payload, {exp: notAfter, iat: nowInSecs});
```

Comment:
Style nit, keep function on same line as equals if possible (and also use `const`), change to:

```js
const payload = _.assign(
  {}, options.payload, {exp: notAfter, iat: nowInSecs});
```